zap/src/main/java/org/zaproxy/zap/extension/ascan/filters/GenericFilterUtility.java

```diff
@@ -0,0 +1,272 @@
+            Collection<T> nodeValues,
+            String filterType) {
+        for (GenericFilterData<T> genericFilterData : genericFilterDataCollection) {
+            FilterCriteria filterCriteria = genericFilterData.getFilterCriteria();
+            switch (filterCriteria) {
+                case INCLUDE:
+                    if (genericFilterData.getValues().isEmpty()) {
+                        return FilterResult.NOT_FILTERED;
+                    }
+                    for (T value : nodeValues) {
+                        if (genericFilterData.getValues().contains(value)) {
+                            return FilterResult.NOT_FILTERED;
+                        }
+                    }
+
+                    return new FilterResult(
+                            Constant.messages.getString(
+                                    INCLUDE_FILTER_CRITERIA_MESSAGE_KEY,
+                                    new Object[] {filterType, genericFilterData.getValues()}));
+                case EXCLUDE:
+                    for (T value : nodeValues) {
+                        if (genericFilterData.getValues().contains(value)) {
+                            return new FilterResult(
+                                    Constant.messages.getString(
+                                            EXCLUDE_FILTER_CRITERIA_MESSAGE_KEY,
+                                            new Object[] {filterType, "[" + value + "]"}));
+                        }
+                    }
+                    return FilterResult.NOT_FILTERED;
+                default:
+                    return FilterResult.NOT_FILTERED;
+            }
+        }
+        return FilterResult.NOT_FILTERED;
+    }
+
+    /**
+     * this method will call {@code GenericFilterUtility#isFiltered(Collection, Collection, String)}
+     * method internally so basic requirement of equals and hasCode for the Classes holds same.
+     *
+     * @param <T> any object implementing hashCode and equals method
+     * @param genericFilterDataCollection Collection of data provided by setting filter criteria
+     * @param nodeValue data associated with the scanned node.
+     * @param filterType type of the filter ie Tag/Http Status code/Method etc
+     * @return FilterResult contains result after applying filter.
+     */
+    public static <T> FilterResult isFiltered(
+            Collection<GenericFilterData<T>> genericFilterDataCollection,
+            T nodeValue,
+            String filterType) {
+        Set<T> nodeValues = new LinkedHashSet<>();
+        nodeValues.add(nodeValue);
+        return isFiltered(genericFilterDataCollection, nodeValues, filterType);
+    }
+
+    /**
+     * This method is used to create {@link TagScanFilter} instance using Tag List for Include and
+     * Exclude {@code FilterCriteria}
+     *
+     * @param incTagList
+     * @param excTagList
+     * @return List of ScanFilter
+     */
+    public static List<ScanFilter> getTagScanFilters(
```

Comment:
> > Actually alert and confidence were not there in my previous PR but in History filter they are there so i though of adding them. Also say me as a user of tool, has configured some alerts as there is some critical code then only running them is useful. Plus incase there are alerts and after fixing those i want to rerun those alerts.
> 
> No doubt that the retest is useful (that's why we have #375 :) but better if it's more straightforward to use.
> 
> > you are right regarding lot of matches but in combination to other filters it can be narrowed down.
> 
> You can achieve that without the confidence/risk, filter by URL would already match just the ones you need.

you are right we can achieve by other ways but it would be difficult say 10 urls are having issues then from usability point, we can just have a filter by alert instead of configuring those urls.

Reply:
didn't got your point. Caller will be FilterPanel but depending on if ScanFilter extends AbstractGenericFilter or not input changes. can you please give some snippet for better understanding,

Reply:
I was suggesting adding something like this:
```Java
public static <V, T extends AbstractGenericFilter<V>> List<ScanFilter> createFilter(Supplier<T> constructor, 
        List<V> incTagList, List<V> excTagList) {
    List<ScanFilter> scanFilterList = new ArrayList<>();
    if (!CollectionUtils.isEmpty(incTagList)) {
        GenericFilterData<V> genericFilterData = new GenericFilterData<>(FilterCriteria.INCLUDE, incTagList);
        T filter = constructor.get();
        filter.getGenericFilterDataCollection().add(genericFilterData);
        scanFilterList.add(filter);
    }

    if (!CollectionUtils.isEmpty(excTagList)) {
        GenericFilterData<V> genericFilterData = new GenericFilterData<>(FilterCriteria.EXCLUDE, excTagList);
        T filter = constructor.get();
        filter.getGenericFilterDataCollection().add(genericFilterData);
        scanFilterList.add(filter);
    }

    return scanFilterList;
}
```

(This indeed assumes that the filters are of `AbstractGenericFilter` type, though it matches with `GenericFilterUtility`.)

Then the callers would do:
```Java
GenericFilterUtility.createFilter(MethodScanFilter::new, new ArrayList<>(), new ArrayList<>());
GenericFilterUtility.createFilter(TagScanFilter::new, new ArrayList<>(), new ArrayList<>());
```

Reply:
hmm this is quite useful. we can call for all the scanFilters extending AbstractgenericFilter. thanks alot.